funnel/extapi/explara.py

```diff
@@ -0,0 +1,62 @@
+# -*- coding: utf-8 -*-
+
+import requests
+from ..util import format_twitter
+from ..models import SyncTicket
+
+__all__ = ['ExplaraAPI']
+
+
+class ExplaraAPI(object):
+    def __init__(self, access_token):
+        self.access_token = access_token
+        self.headers = {'Authorization': u'Bearer ' + self.access_token}
+        self.base_url = 'https://www.explara.com/api/e/{0}'
+
+    def get_orders(self, explara_eventid):
+        ticket_orders = []
+        completed = False
+        from_record = 0
+        to_record = 50
+        while not completed:
+            payload = {'eventId': explara_eventid, 'fromRecord': from_record, 'toRecord': to_record}
+            attendee_response = requests.post(self.base_url.format('attendee-list'), headers=self.headers, data=payload).json()
+            if not attendee_response.get('attendee'):
+                completed = True
+            elif isinstance(attendee_response.get('attendee'), list):
+                ticket_orders.extend([order for order in attendee_response.get('attendee')])
+            # after the first batch, subsequent batches are dicts with batch no. as key.
+            elif isinstance(attendee_response.get('attendee'), dict):
+                ticket_orders.extend([order for order_idx, order in attendee_response.get('attendee').items()])
+            from_record = to_record
+            to_record += 50
+
+        return ticket_orders
+
+    def get_tickets(self, explara_eventid):
+        tickets = []
+        for order in self.get_orders(explara_eventid):
+            for attendee in order.get('attendee'):
+                # cancelled tickets are in this list too, hence the check
+                if attendee.get('status') == 'attending':
+                    # we sometimes get an empty array for details
+                    details = attendee.get('details') or {}
+                    tickets.append({
+                        'fullname': attendee.get('name'),
+                        'email': attendee.get('email'),
+                        'phone': details.get('Phone') or order.get('phoneNo', ''),
+                        'twitter': format_twitter(details.get('Twitter handle', '')),
+                        'job_title': details.get('Job title', ''),
+                        'company': details.get('Company name', ''),
+                        'city': order.get('city', ''),
+                        'ticket_no': attendee.get('ticketNo'),
+                        'ticket_type': attendee.get('ticketName').strip(),
```

Comment:
You should do `.strip()` for other fields too. For anyone registering from a phone, SwiftKey automatically inserts a trailing space, so unless Explara strips it, you should.